### PR TITLE
supersig m3 completion

### PR DESCRIPTION
**Pull request by decentration**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/959 < please fill this in with the PR number of your application.

### CODE REVIEW

**Comment by decentration** [comment]
@0xCaso Hey Matteo, if you're available, M3 is ready for your perusal! Some nice improvements have been made which will greatly improve user experience.

**Comment by 0xCaso** [comment]
Hi @decentration great, and thanks for the delivery! Actually I have a bit of backlog, but for sure I can work on it early next week.

**Comment by decentration** [comment]
sounds good @0xCaso look forward!

File: deliveries/pallet_supersig_decentration-milestone-3.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Milestone Delivery :mailbox:
+
+> ⚡ Only the GitHub account that submitted the application is allowed to submit milestones. 
+> 
+> Don't remove any of the mandatory parts presented in bold letters or as headlines! Lines starting with `>`, such as this one, can be removed.
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).**  
+
+* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/pallet_supersig.md
+* **Milestone Number:** 2
```
Comment (0xCaso) [inline deliveries/pallet_supersig_decentration-milestone-3.md:10]: Milestone 3, not 2
```suggestion
* **Milestone Number:** 3
```

**Review by 0xCaso — COMMENTED** [review]
I started the evaluation :)
In the meantime, could you fix this minor thing in the delivery doc?

**Review by 0xCaso — COMMENTED** [review]
I'm testing the pallet, and I followed the tutorial article and the loom video (which was really clear, thanks!).

Btw, I'm trying to execute proposals that will success (transfer x < supersig balance) and not (>).

In every case, I got the same output, but I've seen in the video that actually you changed this (you can see the screenshots below: proposal 0 was successful, proposal 2 wasn't).

Screenshot 2023-02-27 at 11 33 54

Screenshot 2023-02-27 at 11 33 33

I'm using [substrate-supersig-template](https://github.com/decentration/substrate-supersig-template) (branch `v0.9.28-new`) and [apps](https://github.com/decentration/apps.git) (branch `page-supersig-apps-new`).

I've just noticed that the specified branch of substrate-supersig-template has not been updated recently. Could this be the issue?

**Comment by decentration** [comment]
hey @0xCaso, great stuff i'm glad it was clear!

Re: your above issue:

> I've just noticed that the specified branch of substrate-supersig-template has not been updated recently.

This shouldn't be an issue because the changes are in the `pallet-supersig` and `apps`. However make sure to `cargo update` before recompiling, so that it pulls in the changes from `pallet-supersig`.

Having said that, if you didn't recompile, and the transaction was not a success, then you would see nothing instead of seeing `Ok...`. Ok comes from `CallExecutionAttempted(T::AccountId, CallId, DispatchResultWithPostInfo),` from the [pallet](https://github.com/kabocha-network/pallet_supersig/blob/3dbf22f8b51ffe4d100cb2f8d4bc45b17e621225/src/lib.rs#L189). It's very unlikely that if the transaction was not a success that it would show `Ok` because this is a primitive from substrate's `FRAME`.

There is a chance that the balance you've sent is like 500 / 1_000_000_000_000 instead of 500. I think this is the most likely issue. 

Are you using the docker? There is a chance i may have not pushed the newest version which may have fixed the decimals issue i just mentioned.

**Comment by decentration** [comment]
Screenshot 2023-02-27 at 11 22 17

I just tried to send 5000 from SUPERSIG which only has a balance of 4500, and got the above error. And i am very sure all this is the same version, because i don't think changes have been made recently to affect the decimal issue.

**Comment by decentration** [comment]
Screenshot 2023-02-27 at 11 25 42

@0xCaso Yes it looks like you are on an older version 

Screenshot 2023-02-27 at 11 26 25

The result should display the above whereas your result only shows `<...SpRuntimeDispactchError>`. 

So yes you will need to `cargo update`, sorry that was confusing i should have mentioned, given its on the same release branch!

**Comment by 0xCaso** [comment]
Thanks for the fast response! Yes, the problem was that I didn't do `cargo update` before building. Now I can see the new messages.

**Comment by 0xCaso** [comment]
Also, I'm trying to run `pallet_supersig` tests but I have the following 5 errors, do you have idea of the reason?
```rust
error[E0308]: mismatched types
   --> src/tests/approve_call.rs:113:8
    |
113 |                 Ok(Ok(()))
    |                 -- ^^^^^^ expected struct `hidden_include::dispatch::PostDispatchInfo`, found enum `Result`
    |                 |
    |                 arguments to this enum variant are incorrect
    |
    = note: expected struct `hidden_include::dispatch::PostDispatchInfo`
                 found enum `Result<(), _>`
note: tuple variant defined here
   --> /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/result.rs:508:5

error[E0308]: mismatched types
   --> src/tests/approve_call.rs:181:8
    |
181 |                 Ok(Ok(()))
    |                 -- ^^^^^^ expected struct `hidden_include::dispatch::PostDispatchInfo`, found enum `Result`
    |                 |
    |                 arguments to this enum variant are incorrect
    |
    = note: expected struct `hidden_include::dispatch::PostDispatchInfo`
                 found enum `Result<(), _>`
note: tuple variant defined here
   --> /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/result.rs:508:5

error[E0599]: no variant or associated item named `InvalidNumberOfMembers` found for enum `pallet::Error` in the current scope
   --> src/tests/create_supersig.rs:136:19
    |
136 |             Error::<Test>::InvalidNumberOfMembers
    |                            ^^^^^^^^^^^^^^^^^^^^^^ variant or associated item not found in `pallet::Error<mock::Test>`
    |
   ::: src/lib.rs:202:5
    |
202 |     pub enum Error<T> {
    |     ----------------- variant or associated item `InvalidNumberOfMembers` not found for this enum

error[E0599]: no variant or associated item named `InvalidNumberOfMembers` found for enum `pallet::Error` in the current scope
   --> src/tests/leave_supersig.rs:93:19
    |
93  |             Error::<Test>::InvalidNumberOfMembers
    |                            ^^^^^^^^^^^^^^^^^^^^^^ variant or associated item not found in `pallet::Error<mock::Test>`
    |
   ::: src/lib.rs:202:5
    |
202 |     pub enum Error<T> {
    |     ----------------- variant or associated item `InvalidNumberOfMembers` not found for this enum

error[E0599]: no variant or associated item named `InvalidNumberOfMembers` found for enum `pallet::Error` in the current scope
   --> src/tests/remove_members.rs:112:19
    |
112 |             Error::<Test>::InvalidNumberOfMembers
    |                            ^^^^^^^^^^^^^^^^^^^^^^ variant or associated item not found in `pallet::Error<mock::Test>`
    |
   ::: src/lib.rs:202:5
    |
202 |     pub enum Error<T> {
    |     ----------------- variant or associated item `InvalidNumberOfMembers` not found for this enum
```

**Comment by decentration** [comment]
@0xCaso ah yes, the tests were updated in the later version (0.9.37), i've now pulled them into 0.9.28, just fetch and git pull and try again, and you should have working tests.

**Review by 0xCaso — CHANGES_REQUESTED** [review]
Thanks again, I posted my evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/pallet_supersig_3_0xCaso.md). Basically I just mention to fix a little thing on the Medium guide.

**Review by 0xCaso — APPROVED** [review]
Thanks! And no, it was straightforward ;)

**Review by Noc2 — APPROVED** [review]

**Comment by 0xCaso** [comment]
Thanks again @decentration for the contribution and congrats! I've forwarded internally the invoice, the payment should happen within 14 days.

**Comment by decentration** [comment]
Thank you @0xCaso pleasure to interact with you : )

**Comment by RouvenP** [comment]
hi @decentration we transferred the payment today

**Comment by decentration** [comment]
> hi @decentration we transferred the payment today

Hey Rouven, thanks and received 👍

**Pull request merged**